### ISSUE TITLE
ScrollBox: Inconsistent behaviour with shadow visibility

### ISSUE THREAD

**Issue by Meorawr** [Stanzilla/WoWUIBugs#158]
Not necessarily classing this as a "bug" and more a query at this point.

ScrollBox supports controlling the visibility of two atlases for a "shadow" effect. [The current visibility rules](https://github.com/Gethe/wow-ui-source/blob/fb934dc6558497b2a494898a33e0587a0e6bfe3d/SharedXML/Scroll/ScrollBox.lua#L158) are:

- The top shadow atlas will be shown if the scroll percentage is > 0.
- The bottom shadow atlas will be shown if the region has a scrollable extent.

These two seem inconsistent if the shadow is meant to indicate - in the absence of a scrollbar - whether or not a region can be scrolled further in a given direction. Specifically, the bottom shadow always being visible even while scrolled to the bottom of the region runs counter to my personal expectations.

The expected visibility rules would be as follows:

- The top shadow atlas will be shown if the scroll percentage is > 0.
- The bottom shadow atlas **should** be shown if the region has a scrollable extent _and_ if the scroll percentage is < 1.

A sample is provided below demonstrating this, setting up a black scrollbox with two red atlases for the shadow textures. 

```lua
local DataProvider = CreateDataProviderIndexRange(20);

local ScrollView = CreateScrollBoxListLinearView();
ScrollView:SetDataProvider(DataProvider);
ScrollView:SetElementExtent(28);
ScrollView:SetElementInitializer("Button", "", nop);

local ScrollBox = CreateFrame("Frame", nil, UIParent, "WowScrollBoxList");
ScrollBox:SetPoint("CENTER");
ScrollBox:SetSize(400, 400);
ScrollBox:SetLowerShadowAtlas("colorblind-bar-red", true);
ScrollBox:SetUpperShadowAtlas("colorblind-bar-red", true);

local ScrollBar = CreateFrame("EventFrame", nil, UIParent, "WowTrimScrollBar");
ScrollBar:SetPoint("TOPLEFT", ScrollBox, "TOPRIGHT");
ScrollBar:SetPoint("BOTTOMLEFT", ScrollBox, "BOTTOMRIGHT");

ScrollBox.Background = ScrollBox:CreateTexture(nil, "BACKGROUND");
ScrollBox.Background:SetAllPoints(ScrollBox);
ScrollBox.Background:SetColorTexture(0, 0, 0);

ScrollUtil.InitScrollBoxWithScrollBar(ScrollBox, ScrollBar, ScrollView);
```

The video below demonstrates that the visibility of the top shadow depends upon whether or not the region has been scrolled down, but that the bottom shadow will always be visible - even at the bottom. Note that we can also see a separate bug (#159) where dragging the scrollbar doesn't appear to correctly control the visibility of the top shadow 100% of the time.

https://user-images.githubusercontent.com/287102/128736432-be9bc40e-039a-4fe1-94ac-a278d9f5ac55.mp4

**Comment by Meorawr** [comment]
Fixed in 2.5.2.39801 for BCC; will wait until fixed on retail before closing.

**Comment by Meorawr** [comment]
Fixed in 1.14.0.39958 and 9.1.5.39977.
